### PR TITLE
Fix/load env

### PR DESCRIPTION
**Pull request by Paras-Wednesday**
### Ticket Link
---------------------------------------------------

### Related Links
---------------------------------------------------

### Description
---------------------------------------------------
* fixes the loading db secrets in local env, and adds test case for the same

### Steps to Reproduce / Test
---------------------------------------------------

### Request
---------------------------------------------------

### Response
---------------------------------------------------

## Release Notes

- **New Features**
	- Enhanced build process for improved efficiency in the Go application.
	- New installation command for `go-commitlinter` to validate commit messages.

- **Bug Fixes**
	- Improved error handling and message formatting across various components.

- **Documentation**
	- Updated `README.md` for better clarity and consistency.

- **Tests**
	- Expanded testing coverage for environment configurations and error handling related to database credentials.
	- Standardized error message formatting in tests for consistency.

### CODE REVIEW

File: internal/config/env_test.go [hunk 1]
```diff
@@ -2,11 +2,12 @@ package config_test
 
 import (
 	"fmt"
-	. "go-template/internal/config"
 	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+
+	. "go-template/internal/config"
```
[the remark this answers is not in the document]
Comment (Paras-Wednesday) [reply internal/config/env_test.go:10]: Will be removed going forward, currently not possible, as will require big refactor

File: internal/config/env.go [hunk 2]
```diff
@@ -71,45 +75,61 @@ func LoadEnv() error {
...
+		return fmt.Errorf("couldn't unmarshal db secret: %w", err)
 	}
-	return fmt.Errorf("COPILOT_DB_CREDS_VIA_SECRETS_MANAGER should have had a value")
+
+	os.Setenv("PSQL_DBNAME", secrets.DBName)
+	os.Setenv("PSQL_HOST", secrets.Host)
+	os.Setenv("PSQL_PASS", secrets.Password)
+	os.Setenv("PSQL_PORT", strconv.Itoa(secrets.Port))
+	os.Setenv("PSQL_USER", secrets.Username)
+
```
[the remark this answers is not in the document]
Comment (Paras-Wednesday) [reply internal/config/env.go:133]: In the context, of the file. This is good enough

File: internal/config/env.go [hunk 3]
```diff
@@ -57,9 +58,12 @@ func FileName() string {
 }
 
 func LoadEnv() error {
+	const (
+		localEnvFile = "local"
```
Comment (Anurag-Wednesday) [inline internal/config/env.go:62]: should this be named "localEnvName"?

Comment (Paras-Wednesday) [reply internal/config/env.go:62]: Yeah seems more appropriate, I will change it

**Review by Anurag-Wednesday — APPROVED** [review]

File: internal/config/env.go (hunk 3 above)
Comment (Paras-Wednesday) [reply internal/config/env.go:62]: I feel the more appropriate name would be `localEnvName`

**Pull request merged**